src/torchmetrics/detection/mean_ap.py

```diff
@@ -329,6 +332,7 @@ def __init__(
         max_detection_thresholds: Optional[List[int]] = None,
         class_metrics: bool = False,
         extended_summary: bool = False,
+        average: Literal["macro", "micro"] = "macro",
```

Comment:
Thanks for the suggestion. I agree that it would probably be a good idea exchanging literal with enum 
As we have multiple uses literal in this class, I probably wait with this kind of change to a follow up PR and change all occurrences.